### PR TITLE
Export EXIT_CODE from the user provided command, and propagate the error to Cloud Console UI

### PR DESCRIPTION
**Pull request by Obliviour**
- Refactor command argument passthrough
- Spelling
- Enable EXIT_CODE to be of the user provided command
- Move appending to command / internal command to its own function
- Remove outdated arguments from workload create
- Propagate error message to cloud consule

## Fixes / Features
- EXIT_CODE to be linked to the user provided function

### Is Happy workload successful with EXIT_CODE=0
 python3 xpk.py workload create --cluster vbarr-v4-xpk-test-tpu   --workload vbarr-xpk-error-test-2 --command "bash test.sh"   --tpu-type=v4-8 --num-slices=1 --zone=us-central2-b --project=cloud-tpu-multipod-dev
image

### Error workload with EXITO_CODE=1
 python3 xpk.py workload create --cluster vbarr-v4-xpk-test-tpu   --workload vbarr-xpk-error-test-fail --command "bash test.sh && ech a"  --tpu-type=v4-8 --num-slices=1 --zone=us-central2-b --project=cloud-tpu-multipod-dev
image

### XLA Flag work?

 python3 xpk.py workload create --cluster vbarr-v4-xpk-test-tpu   --workload vbarr-xpk-error-test-fail --command "bash test.sh && ech a" --debug-dump-gcs=some_bucket --enable-debug-logs  --tpu-type=v4-8 --num-slices=1 --zone=us-central2-b --project=cloud-tpu-multipod-dev
image

### Does the error propagate to pantheon UI?
image

- [ y] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: xpk.py [hunk 1]
```diff
@@ -2126,7 +2136,7 @@ def get_main_container(args, system, docker_image, command, resource_type) -> st
                 - bash
                 - -c
                 - |
-                  echo XPK Start: $(date) ; _sigterm() ( kill -SIGTERM $!;); trap _sigterm SIGTERM; ({command}) & PID=$!; while kill -0 $PID 2>/dev/null; do sleep 5; done; EXIT_CODE=$? ; echo XPK End: $(date); echo EXIT_CODE=$EXIT_CODE
+                  echo XPK Start: $(date) ; _sigterm() ( kill -SIGTERM $!;); trap _sigterm SIGTERM; ({command}) & PID=$!; while kill -0 $PID 2>/dev/null; do sleep 5; done; wait $PID; EXIT_CODE=$? ; {xpk_internal_commands} echo XPK End: $(date); echo EXIT_CODE=$EXIT_CODE; exit $EXIT_CODE
```
Comment (rwitten) [inline xpk.py:2139]: do we want a semicolon after {xpk_internal_commands}?

File: xpk.py [hunk 2]
```diff
@@ -2101,18 +2100,29 @@ def get_main_and_sidecar_container(args, system, docker_image, command) -> str:
...
   """
+
+  xpk_internal_commands = ""
+  if args.debug_dump_gcs:
+    xpk_internal_commands += ('WORKER_ID=$HOSTNAME;'
+                f'gsutil cp -r /tmp/xla_dump/ {args.debug_dump_gcs}/$WORKER_ID;')
+
+  command = args.command
+  if args.enable_debug_logs:
+    command = ('TPU_STDERR_LOG_LEVEL=0 TPU_MIN_LOG_LEVEL=0 TF_CPP_MIN_LOG_LEVEL=0'
```
Comment (rwitten) [inline xpk.py:2123]: nit: probably we should have prequel command set at some point hahah :-)

**Review by rwitten — APPROVED** [review]

File: xpk.py (hunk 2 above)
Comment (Obliviour) [reply xpk.py:2123]: Agreed, lets add this later when needed.

For now `enable_debug_logs` needs the environmental variables to be within the same background process executed within ({command}). Other prequel commands might want to exist outside the background process so I will wait for those new use cases before adding the prequel potentially incorrectly now.

File: xpk.py (hunk 1 above)
Comment (Obliviour) [reply xpk.py:2139]: I moved the semicolon within the command appended by xpk_internal_commands to be after {xpk_internal_commands}

**Pull request merged**